core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts

```diff
@@ -674,4 +676,16 @@ browse from the page internal links
             this.pageLanguageId = params['language_id'];
         });
     }
+
+    /**
+     * Replaces the value of "_content_lang" in the given string with a new value.
+     * @param {string} inputString - The original string containing the "_content_lang" parameter.
+     * @returns {string} - The updated string with the replaced "_content_lang" value.
+     */
+    private setCurrentContentLang(inputString: string): string {
+        const regex = /(_content_lang=)(\d+)(?=&|$)/;
+        const replacement = `$1${this.pageLanguageId}`;
+
+        return inputString.replace(regex, replacement);
+    }
```

Comment:
Would it be safer to utilize the URL API and replace the query parameters with a URL?

```ts
    private setCurrentContentLang(url: string): string {
          const newUrl = new URL(url);
          newUrl.searchParams.set('_content_lang', this.pageLanguageId);
          return newUrl.toString()
    }
```

Reply:
Safer.

Reply:
done.